specification.md

```diff
@@ -1059,6 +1061,140 @@ Note that the referenced function definition type in this case must be `rest` (d
 
 For more information about functions, reference the [Functions definitions](#Function-Definition) section.
 
+#### Using functions for HTTP Service Invocations
+
+For a few reasons, workflow designers might see necessary making straight HTTP invocations to other systems, such as:
+
+1. Target service might not expose an OpenAPI specification descriptor;
+2. Target service might not implement RESTful interfaces;
+3. The use case requires a simple HTTP request invocation.
+
+The specification supports describing HTTP invocations via a simplified interface in the [functions definition](#Function-Definition).
+
+The HTTP invocation interface can be described in the `operation` attribute of the function definition, as shown in the example:
+
+```json
+{
+"functions": [
+  {
+    "name": "queryUserById",
+    "operation": "GET#/users/{id}",
```

Comment:
would it be ok to not use relative url
i assume since type is http that this does not support https, so we could say that "http:// is pre-pended always 
and operation could be like "GET#myapp.com/users{id}"

Reply:
Agreed. How is the scheme specified (most relevantly "http:" vs. "https:")? How is the host specified?

Reply:
Sorry, I disagree. 

> Agreed. How is the scheme specified (most relevantly "http:" vs. "https:")? How is the host specified?

Implementations would instruct users on how to properly configure their endpoints per function. In a real use case scenario, these endpoints would change when deploying the workflow on different envs. 

I wouldn't expect users to hardcode the URLs here. The path is a description of the operation, not the full endpoint 

So, for instance:

```json
"functions": [
  {
    "name": "queryUserById",
    "operation": "GET#/users/{id}",
    "type": "http"
  } 
]
```

Describes that this function, in the **path** `/users/{id}` via the GET method would perform a read operation in the users' domain.

Implementations can find a way to provide an external configuration for users to reliably set up their deployments across environments without having to change the operation definition.

For example: `my.configuration.workflowid.funcid.host=http://myservice.com`.

Reply:
Then why not point inside a _file_ expressing such configuration, as is done for
* [OpenAPI](https://github.com/serverlessworkflow/specification/blob/main/specification.md#using-functions-for-restful-service-invocations) (`"operation": "file://confirmationapi.json#sendOrderConfirmation"`), and
* [Async API](https://github.com/serverlessworkflow/specification/blob/main/specification.md#using-functions-for-async-api-service-invocations) (`"operation": "file://streetlightsapi.yaml#onLightMeasured"`), and
* [gRPC](https://github.com/serverlessworkflow/specification/blob/main/specification.md#using-functions-for-rpc-service-invocations) (`"operation": "file://myuserservice.proto#UserService#ListUsers"`), and
* [GraphQL](https://github.com/serverlessworkflow/specification/blob/main/specification.md#using-functions-for-graphql-service-invocations) (`"operation": "https://example.com/pets/graphql#query#pet"`), and
* [OData](https://github.com/serverlessworkflow/specification/blob/main/specification.md#using-functions-for-odata-service-invocations) (`"operation": "https://services.odata.org/V3/OData/OData.svc#Persons"`)

?

Reply:
yes, making it similar would to what we already have would be preferred pls

Reply:
still tho like been saying imo users should be able to define a single function def for all http calls which then is rly same thing as what we have already with custom function type. ill show example what i mean in meet

Reply:
i think we can handle all non-spec-compliant invocations by nust updating custom function type a little